docs/maps/clean-data.asciidoc

```diff
@@ -0,0 +1,183 @@
+----
+
+
+[float]
+==== Simplifying region datasets
+
+Region datasets are polygon datasets where the boundaries of the documents don't overlap. This is common for administrative boundaries, land usage, and other continuous datasets. This type of datasets has the special feature that any geospatial operation modifying the lines of the polygons needs to be applied in the same way to the common sides of the polygons to avoid the generation of thin gap and overlap artifacts. 
+
+https://github.com/mbloch/mapshaper[`mapshaper`] is an excellent tool to work with this type of datasets as it understands datasets of this nature and works with them accordingly.
+
+Depending on the usage of a region dataset, different geospatial precisions may be adequate. A world countries dataset that is displayed for the entire planet does not need the same precision as a map of the countries in the South Asian continent.
+
+`mapshaper` offers a https://github.com/mbloch/mapshaper/wiki/Command-Reference#-simplify[`simplify`] command that accepts percentages, resolutions, and different simplification algorithms.
+
+[source,sh]
+----
+# Example NAD83 file from https://www2.census.gov/geo/tiger/GENZ2018/shp/cb_2018_us_county_5m.zip
+#
+# Generate a baseline GeoJSON file from OGR
+$ ogr2ogr \
+  -f GeoJSON "cb_2018_us_county_5m.ogr.geo.json" \
+  -t_srs "EPSG:4326" \
+  -lco RFC7946=YES \
+  "cb_2018_us_county_5m.shp" \
+  "cb_2018_us_county_5m"
+
+# Simplify at different percentages with mapshaper
+$ for pct in 10 50 75 99; do \
+  mapshaper \
+    -i "cb_2018_us_county_5m.shp" \ # Input file
+    -proj "EPSG:4326" \ # Output projection
+    -simplify "${pct}%" \ # Simplification
+    -o cb_2018_us_county_5m.mapshaper_${pct}.geo.json; \ # Output file
+  done
+
+# Compare the size of the output files
+$ du -h cb_2018_us_county_5m*.geo.json
+2,0M	cb_2018_us_county_5m.mapshaper_10.geo.json
+4,1M	cb_2018_us_county_5m.mapshaper_50.geo.json
+5,3M	cb_2018_us_county_5m.mapshaper_75.geo.json
+6,7M	cb_2018_us_county_5m.mapshaper_99.geo.json
+6,7M	cb_2018_us_county_5m.ogr.geo.json
+----
+
+
+[float]
+==== Fixing incorrect geometries
+
+The Maps application expects valid GeoJSON or Shapefile datasets. Apart from the mentioned CRS requirement, also geometries inside the dataset need to be valid. Both `ogr2ogr` and `mapshaper` have options to try to fix invalid geometries:
+
+* OGR https://gdal.org/programs/ogr2ogr.html#cmdoption-ogr2ogr-makevalid[`-makevalid`] option
+* Mapshaper https://github.com/mbloch/mapshaper/wiki/Command-Reference#-clean[`-clean`] command
+
+
+[float]
+==== Conclusion
+
+Both tools are excellent geospatial ETL (Extract Transform and Load) utilities that can do much more than viewed here. Reading the documentation in detail is worth investment to improve the quality of the datasets by removing unwanted fields, refining data types, validating value domains, etc. Finally, being command line utilities, both can be automated and added to QA pipelines.
```

Comment:
How about replacing "Both tools are" with "`ogr2ogr` and `mapshaper` are". I think more context is better since users may skip to this section and not understand what "Both" refers to.

Reply:
Indee, thanks!